tests/src/Kernel/DatabaseLogTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\rdf_entity\Kernel;
+
+use Drupal\Core\Database\Database;
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\Tests\rdf_entity\Traits\RdfDatabaseConnectionTrait;
+
+/**
+ * Tests the query logging.
+ *
+ * @group rdf_entity
+ *
+ * @coversDefaultClass \Drupal\rdf_entity\Database\Driver\sparql\Connection
+ */
+class DatabaseLogTest extends KernelTestBase {
+
+  use RdfDatabaseConnectionTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+    $this->setUpSparql();
+  }
+
+  /**
+   * Tests the log.
+   *
+   * @param string $method
+   *   The method name 'query' or 'update'.
+   * @param string $query
+   *   The query.
+   * @param array $args
+   *   The query arguments.
+   *
+   * @dataProvider provider
+   */
+  public function testLog(string $method, string $query, array $args): void {
+    Database::startLog('log_test', 'sparql_default');
+    $this->sparql->{$method}($query);
+    $log = $this->sparql->getLogger()->get('log_test');
```

Comment:
`$args` is never passed to the connection handler? Are these perhaps being passed directly in the query?

I'm not sure of the implications of this in Sparql, but in SQL the arguments are passed separately so they can be escaped to prevent injection attacks. Isn't this a concern in Sparql?

Reply:
There's TODO to add arguments in the future (not an easy task)

Reply:
I'm passing now `$args` to the query.